api/scpca_portal/models/computed_file.py

```diff
@@ -141,42 +157,73 @@ def get_project_file(cls, project, download_config: Dict, computed_file_name: st
         # If the query return empty, then an error occurred, and we should abort early
         if not libraries.exists():
             return
+
         libraries_metadata = [
             lib for library in libraries for lib in library.get_combined_library_metadata()
         ]
-        metadata_path_var = (
-            f'{download_config["modality"]}_METADATA_FILE_NAME'
-            if not download_config["metadata_only"]
-            else "METADATA_ONLY_FILE_NAME"
-        )
-        metadata_file.write_metadata_dicts(
-            libraries_metadata, getattr(cls.MetadataFilenames, metadata_path_var)
-        )
+        local_metadata_path = ComputedFile.get_local_project_metadata_path(project, download_config)
+        metadata_file.write_metadata_dicts(libraries_metadata, local_metadata_path)
 
         library_data_file_paths = [
-            fp for lib in libraries for fp in lib.get_filtered_data_file_paths(download_config)
+            fp for lib in libraries for fp in lib.get_download_config_file_paths(download_config)
         ]
-        project_data_file_paths = project.get_filtered_data_file_paths(download_config)
+        project_data_file_paths = project.get_download_config_file_paths(download_config)
 
         zip_file_path = common.OUTPUT_DATA_PATH / computed_file_name
         with ZipFile(zip_file_path, "w") as zip_file:
             # Readme file
-            zip_file.write(*(ComputedFile.get_readme_from_download_config(download_config)))
+            zip_file.write(
+                ComputedFile.get_readme_from_download_config(download_config),
+                cls.OUTPUT_README_FILE_NAME,
+            )
             # Metadata file
-            zip_file.write(getattr(cls.MetadataFilenames, metadata_path_var))
+            output_file_constant = (
+                f'{download_config["modality"]}_METADATA_FILE_NAME'
+                if not download_config["metadata_only"]
+                else "METADATA_ONLY_FILE_NAME"
+            )
+            zip_file.write(
+                local_metadata_path, getattr(cls.MetadataFilenames, output_file_constant)
+            )
 
             if not download_config.get("metadata_only", False):
                 for file_path in library_data_file_paths:
-                    zip_file.write(file_path)
-
-            for file_path in project_data_file_paths:
-                zip_file.write(file_path)
+                    output_path = file_path.relative_to(f"{project.scpca_id}/")
+                    # Swap delimiter in multiplexed sample libraries from comma to underscore
+                    if "," in file_path.parent.name:
+                        new_delimiter_dir = Path("_".join(file_path.parent.name.split(",")))
+                        output_path = new_delimiter_dir / file_path.name
+                    zip_file.write(Library.get_local_file_path(file_path), output_path)
```

Comment:
```suggestion
                    zip_file.write(
                        Library.get_local_file_path(file_path),
                        Library.get_zip_file_path(file_path),
                    )
```

Then make this method abstract and just pop off the "root" of the filepath and then rename any comma separated folders or files in the returned path.

```python

@classmethod
def get_zip_file_path(cls, file_path: Path) -> Path:
    # we don't want the file nested in the project directory
    output_path = # remove the parent path
    # we want to only output _ delimited files and folders
    # rename any folder that is comma separated  
    return new_delimiter_dir / file_path.name
```